tests/common/plugins/conditional_mark/__init__.py

```diff
@@ -0,0 +1,295 @@
+"""Plugin for adding any mark to specified test cases based on conditions in a centralized file.
+
+This plugin supports adding any mark to specified test cases based on conditions. All the information of test cases,
+marks, and conditions can be specified in a centralized file.
+"""
+import imp
+import json
+import logging
+import os
+import re
+import subprocess
+import yaml
+
+import pytest
+
+from issue import check_issues
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_CONDITIONS_FILE = 'common/plugins/conditional_mark/tests_mark_conditions.yaml'
+
+
+def pytest_addoption(parser):
+    """Add option for specifying your own mark conditions definition file.
+
+    If this option is not specified, the default mark conditions file will be used.
+    """
+    parser.addoption(
+        '--mark-conditions-file',
+        dest='mark_conditions_file',
+        default='',
+        help="Enable DUT hardware resources monitoring")
+
+
+def load_conditions(session):
+    """Load the content from mark conditions file
+
+    Args:
+        session (obj): The pytest session object.
+
+    Returns:
+        dict or None: Return the mark conditions dict or None if there something went wrong.
+    """
+    conditions_file = session.config.option.mark_conditions_file
+    if not conditions_file:
+        conditions_file = DEFAULT_CONDITIONS_FILE
+
+    if not os.path.exists(conditions_file):
+        # No conditions file supplied, skip adding conditional marks
+        return None
+
+    try:
+        with open(conditions_file) as f:
+            logger.debug('Loaded tests skip conditions from {}'.format(conditions_file))
+            return yaml.safe_load(f)
+    except Exception as e:
+        logger.error(repr(e))
```

Comment:
Is it a little too verbose here to log the error twice?

Reply:
Nice catch!